nlmod/gwf/recharge.py

```diff
@@ -163,15 +200,308 @@ def ds_to_evt(gwf, ds, pname="evt", nseg=1, surface=None, depth=None, **kwargs):
+        specified PET rate while the water content (THETA) is greater than the ET
+        extinction water content (EXTWC). The default is False.
+    unsat_etae : bool, optional
+        If True, ET in the unsaturated zone will be simulated using a capillary pressure
+        based formulation. Capillary pressure is calculated using the Brooks-Corey
+        retention function. The default is False.
+    obs_depth_interval : float, optional
+        The depths at which observations of the water depth in each cell are added. The
+        user-specified depth must be greater than or equal to zero and less than the
+        thickness of GWF cellid (TOP - BOT).
+        The
+    ** kwargs : dict
+        Kwargs are passed onto flopy.mf6.ModflowGwfuzf
+
+
+    Returns
+    -------
+    uzf : flopy.mf6.ModflowGwfuzf
+        Unsaturated zone flow package for Modflow 6.
+    """
+    if mask is None:
+        mask = ds["area"] > 0
+
+    if "layer" not in mask.dims:
+        mask = mask.expand_dims(dim={"layer": ds.layer})
+
+    # only add uzf-cells in active cells
+    idomain = get_idomain(ds)
+    mask = mask & (idomain > 0)
+
+    # generate packagedata
+    surfdep = _get_value_from_ds_datavar(ds, "surfdep", surfdep, return_da=True)
+    vks = _get_value_from_ds_datavar(ds, "vk", vks, return_da=True)
+    thtr = _get_value_from_ds_datavar(ds, "thtr", thtr, return_da=True)
+    thts = _get_value_from_ds_datavar(ds, "thts", thts, return_da=True)
+    thti = _get_value_from_ds_datavar(ds, "thti", thti, return_da=True)
+    eps = _get_value_from_ds_datavar(ds, "eps", eps, return_da=True)
+
+    nuzfcells = int(mask.sum())
+    cellids = np.where(mask)
+
+    iuzno = xr.full_like(ds["botm"], -1, dtype=int)
+    iuzno.data[mask] = np.arange(nuzfcells)
+
+    if landflag is None:
+        landflag = xr.full_like(ds["botm"], 0, dtype=int)
+        # set the landflag in the top layer to 1
+        landflag[get_first_active_layer_from_idomain(idomain)] = 1
+
+    # determine ivertcon, by setting its value to iuzno of the layer below
+    ivertcon = xr.full_like(ds["botm"], -1, dtype=int)
+    ivertcon.data[:-1] = iuzno.data[1:]
+    # then use bfill to accont for inactive cells in the layer below, and set nans to -1
+    ivertcon = ivertcon.where(ivertcon >= 0).bfill("layer").fillna(-1).astype(int)
+
+    # packagedata : [iuzno, cellid, landflag, ivertcon, surfdep, vks, thtr, thts, thti, eps, boundname]
+    packagedata = cols_to_reclist(
+        ds,
+        cellids,
+        iuzno,
+        landflag,
+        ivertcon,
+        surfdep,
+        vks,
+        thtr,
+        thts,
+        thti,
+        eps,
+        cellid_column=1,
+    )
+
+    # add perioddata for all uzf cells that are at the surface
+    mask = landflag == 1
+
+    # perioddata : [iuzno, finf, pet, extdp, extwc, ha, hroot, rootact, aux]
+    finf_name_arr, uzf_unique_dic = _get_unique_series(ds, finf, "finf")
+    finf = "rch_name"
```

Comment:
Is this an internal data variable? One that is set by nlmod functions? Maybe we should consider some convention for naming these internal data variables, to make it clear this is something nlmod has created and needs for building models? 

Are there any other similar data vars?

Reply:
I copied the approach from RCH and EVT. However, I think we should not add these variables to ds. I will make an issue about this.